flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/metrics/MetricStore.java

```diff
@@ -82,18 +83,52 @@ synchronized void retainTaskManagers(List<String> activeTaskManagers) {
      */
     synchronized void retainJobs(List<String> activeJobs) {
         jobs.keySet().retainAll(activeJobs);
-        currentExecutionAttempts.keySet().retainAll(activeJobs);
+        representativeAttempts.keySet().retainAll(activeJobs);
     }
 
     public synchronized void updateCurrentExecutionAttempts(Collection<JobDetails> jobs) {
-        jobs.forEach(
-                job ->
-                        currentExecutionAttempts.put(
-                                job.getJobId().toString(), job.getCurrentExecutionAttempts()));
+        for (JobDetails job : jobs) {
+            String jobId = job.getJobId().toString();
+            Map<String, Map<Integer, List<Integer>>> currentExecutionAttempts =
+                    job.getCurrentExecutionAttempts();
+            Map<String, Map<Integer, Integer>> jobRepresentativeAttempts =
+                    new HashMap<>(currentExecutionAttempts.size());
+            for (Map.Entry<String, Map<Integer, List<Integer>>> vertexEntry :
+                    currentExecutionAttempts.entrySet()) {
+                String vertexId = vertexEntry.getKey();
+                Map<Integer, List<Integer>> subtaskAttempts = vertexEntry.getValue();
+                Map<Integer, Integer> vertexRepresentativeAttempts =
```

Comment:
We may simplify the computation to be 

```

Map<Integer, Integer> vertexRepresentativeAttempts =
                        vertexEntry.getValue().entrySet().stream()
                                .filter(entry -> !entry.getValue().isEmpty())
                                .collect(
                                        Collectors.toMap(
                                                Map.Entry::getKey,
                                                entry -> entry.getValue().get(0)));
```
So that we could save some Entry declaration, which makes the code a bit complex.